statics/js/components/inject-form.js

```diff
@@ -217,16 +217,19 @@ Vue.component('inject-form', {
     getIP: function(node) {
       md = node.metadata;
       ipFamily = this.type.slice(-1);
-      if (ipFamily == "4" && "IPV4" in md) {
-        return md.IPV4[0];
-      } else if (ipFamily == "6" && "IPV6" in md) {
-        return md.IPV6[0];
+      if (ipFamily == "4") {
+        if (md.Neutron && md.Neutron.IPV4) return md.Neutron.IPV4[0];
+        if (md.IPV4) return md.IPV4[0];
+      } else if (ipFamily == "6") {
+        if (md.Neutron && md.Neutron.IPV6) return md.Neutron.IPV6[0];
+        if (md.IPV6) return md.IPV6[0];
       } else {
         return "";
       }
     },
 
     getMAC: function(node) {
+      if (node.metadata.ExtID && node.metadata.ExtID.attached-mac) return node.metadata.ExtID.attached-mac;
```

Comment:
You bracket notation (as attribute name contains a hyphen):

```
node.metadata.ExtID["attached-mac"]
```

Else hyphen is interpreted as a minus

Reply:
Thanks for the review Aidan, Fixed!